### PR TITLE
Brass BEDPE

### PR DESCRIPTION
**Pull request by mp15**
Adds support for BEDPE that use commented headers. Incorporates #152.

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

File: variantworks/io/bedio.py [hunk 1]
```diff
@@ -32,7 +34,7 @@ class BEDType(Enum):
         BED = 0
         BEDPE = 1
 
-    def __init__(self, bed_path, bed_type):
+    def __init__(self, bed_path, bed_type, header_is_comment=False):
```
Comment (tijyojwad) [inline variantworks/io/bedio.py:37]: could you add documentation for this new arg to the doc string below?

File: variantworks/io/bedio.py [hunk 2]
```diff
@@ -48,9 +50,42 @@ def __init__(self, bed_path, bed_type):
         self._bed_path = bed_path
         assert(isinstance(bed_type, self.BEDType)), "bed_type must be of BEDType enum."
         self._bed_type = bed_type
-        self._df = pd.read_csv(self._bed_path, delimiter="\t")
+        self._header_is_comment = header_is_comment
+        if self._header_is_comment is True:
+            self._df = self._read_data(self._bed_path)
```
Comment (tijyojwad) [inline variantworks/io/bedio.py:55]: This is a pretty nice change. I'm thinking of making this a standard pre-processing for all BED files, so we don't have to maintain a separate path in the code. Does this work for the typical BED files without a header section? If so, then I would just remove the `header_is_comment` conditional and call the `_read_data` function directly on all inputs. The `uniquify` function might do some extra work, but it'll be quite minimal.

Comment (tijyojwad) [reply variantworks/io/bedio.py:55]: and we could add to the docs how non-unique header names are handled, i.e. with a counter suffix by default

**Review by tijyojwad — CHANGES_REQUESTED** [review]

File: variantworks/io/bedio.py (hunk 1 above)
Comment (mp15) [reply variantworks/io/bedio.py:37]: Done

File: variantworks/io/bedio.py (hunk 2 above)
Comment (mp15) [reply variantworks/io/bedio.py:56]: Alas the BEDPE implementations are a bit fast and loose when it comes to whether or not the header is a comment or not which is why I created a second code path.

Comment (tijyojwad) [reply variantworks/io/bedio.py:56]: Oh that's a bummer. In that case the change looks good to me! I'll add a test case for this code path using the file you linked [here](https://github.com/clara-parabricks/VariantWorks/issues/144#issuecomment-733039660) once this PR goes in

**Review by tijyojwad — APPROVED** [review]

**Pull request merged**